src/frequenz/sdk/actor/power_distributing/power_distributing.py

```diff
@@ -558,25 +558,76 @@ def _get_components_data(self, batteries: Set[int]) -> List[InvBatPair]:
-                battery_id
-            ].peek()
-
-            if battery_data is None:
-                _logger.warning("None returned from battery receiver %d.", battery_id)
-                continue
-
-            inverter_data: Optional[InverterData] = self._inverter_receivers[
-                inverter_id
-            ].peek()
-
-            if inverter_data is None:
-                _logger.warning("None returned from inverter receiver %d.", inverter_id)
+            data = self._get_battery_inverter_data(battery_id, inverter_id)
+            if data is None:
+                _logger.warning(
+                    "Skipping battery %d because its message isn't correct.",
+                    battery_id,
+                )
                 continue
 
-            pairs_data.append(InvBatPair(battery_data, inverter_data))
+            pairs_data.append(data)
         return pairs_data
 
+    def _get_battery_inverter_data(
+        self, battery_id: int, inverter_id: int
+    ) -> Optional[InvBatPair]:
+        battery_data = self._battery_receivers[battery_id].peek()
+        inverter_data = self._inverter_receivers[inverter_id].peek()
+
+        # It means that nothing has been send on this channels, yet.
+        # This should be handled by BatteryStatus so it should never happen.
+        if battery_data is None or inverter_data is None:
+            return None
+
+        not_replaceable_metrics = [
+            battery_data.soc,
+            battery_data.soc_lower_bound,
+            battery_data.soc_upper_bound,
+            # We could replace capacity with 0, but it won't change distribution.
+            # This battery will be ignored anyway.
+            battery_data.capacity,
+        ]
+        if any(map(isnan, not_replaceable_metrics)):
+            _logger.warning("Some metrics for battery %d are NaN", battery_id)
+            return None
+
+        replaceable_metrics = [
+            battery_data.power_lower_bound,
+            battery_data.power_upper_bound,
+            inverter_data.active_power_lower_bound,
+            inverter_data.active_power_upper_bound,
+        ]
+
+        # If all values are ok then return them
+        if all(map(lambda x: not isnan(x), replaceable_metrics)):
+            return InvBatPair(battery_data, inverter_data)
+
+        # Replace NaN with the value of the adjacent component.
+        # If both metrics are None, return None to ignore this battery.
+        replaceable_pairs = [
+            ("power_lower_bound", "active_power_lower_bound"),
+            ("power_upper_bound", "active_power_upper_bound"),
+        ]
+
+        battery_new_metrics = {}
+        inverter_new_metrics = {}
+        for bat_attr, inv_attr in replaceable_pairs:
+            bat_bound = getattr(battery_data, bat_attr)
+            inv_bound = getattr(inverter_data, inv_attr)
+            if isnan(bat_bound) and isnan(inv_bound):
+                _logger.warning("Some metrics for battery %d are NaN", battery_id)
```

Comment:
Same comment about confirming if we really want this to be a warning.

Reply:
We don't. Changed to debug.